### ISSUE TITLE
Prevent Gaia outages from breaking app authentication

### ISSUE THREAD

**Issue by 314159265359879** [blockstack/blockstack-app#3564]
I am getting multiple reports that users are having trouble connecting to dapps with their wallet. After selecting the account to connect with. It loads forever and never logs in on the dapp. 

When doing the same on a ledger enabled wallet extension their is no issue. 

On Firefox I see this:
https://user-images.githubusercontent.com/33360391/232696153-7c3b8bda-49af-45ad-a64c-50ff56cd37c2.mp4

Chrome shows this in console:
![image](https://user-images.githubusercontent.com/33360391/232696480-14166232-99f4-4669-9d73-ed1b96613f0e.png)

**Comment by 314159265359879** [comment]
It looks like this is the underlying issue, already resolved:
![image](https://user-images.githubusercontent.com/33360391/232697980-bb4ccf0d-8808-40c8-aa3d-425848a17875.png)

**Comment by markmhendrickson** [comment]
@kyranjamie should we reduce or remove our Gaia dependency further here so future outages aren't breaking?

**Comment by friedger** [comment]
@markmhx Gaia dependency is part of auth protocol. You want to change the protocol?

**Comment by kyranjamie** [comment]
Can you elaborate on that @friedger? Where does it say this?

**Comment by friedger** [comment]
@kyranjamie We store app meta data in the profile and we use the gaia protocol for storage. See also (https://github.com/stacksgov/sips/blob/58aae536637ad9236365d0873f07d8d831b4add1/sips/sip-x/sip-x-authentication-protocol.md#public-profile)

To remove dependency we either 
A) need to remove the app meta data and public profiles (like for ledger users) or 
B) find a different storage solution.

For A), the auth response could remove the app private key and all the apps needs to handle the reduced auth response.
For a different storage solution B), we need to add support for users to update their zonefile.

**Comment by 314159265359879** [comment]
Update from diwaker
_diwaker — Today 10 minutes ago
Hey all, appreciate the pings -- I'm in EST so appreciate your patience while our team comes online. We're looking into the Gaia hub issue, I don't see anything obviously wrong on our end -- it's likely something with Azure. Please bear with us, we'll update as soon as we have something to share._

**Comment by 314159265359879** [comment]
https://status.hiro.so/
to keep an eye on the current status

**Comment by diwakergupta** [comment]
> https://status.hiro.so/ to keep an eye on the current status

Resolved as of an hour ago.

**Comment by markmhendrickson** [comment]
> To remove dependency we either
> A) need to remove the app meta data and public profiles (like for ledger users) or
> B) find a different storage solution.

@friedger my sense here is that it'd be most expedient to implement option A only for when the wallet detects that Gaia is down, defaulting to a degraded yet unblocked experience a la the one Ledger users have?

**Comment by friedger** [comment]
I agree in the hope that some resources re available to make gaia usable and configurable so that Hiro carries all the load. See forum post about gaia at home.

**Comment by 314159265359879** [comment]
Apparently this error can also be caused by an ad blocker sometimes. Having a solution so that the user can proceed, in degraded state, rather than being blocked completely seems like a good idea. hz.btc agrees ;-)

![image](https://github.com/hirosystems/wallet/assets/33360391/07fe8937-aee1-4e72-b9c4-28c607e952f2)

**Comment by 314159265359879** [comment]
Is gaia always out these days? Seeing this a lot and it prevents the user from connecting to dapps

![Image](https://github.com/user-attachments/assets/b0db7910-b1fe-4f45-9b5d-995eda2aa2b9)

**Comment by 314159265359879** [comment]
This comment related (the rest of the issue seems different, although I am not entirely sure): https://github.com/leather-io/extension/issues/5919#issuecomment-2497548908
